plugins/by-name/codeium-nvim/default.nix

```diff
@@ -15,108 +21,56 @@ lib.nixvim.plugins.mkNeovimPlugin {
     khaneliman
   ];
 
+  description = ''
+    By default, enabling this plugin will also install the `curl`, `gzip`, `coreutils`, `util-linux` and `codeium` packages (via the `dependencies.*.enable` options`)
+
+    Options under `settings.tools.*` with a known dependency will be configured automatically when the associated `dependencies.*.enable` option is enabled.
+
+    You are free to configure `dependencies.*.enable` and `dependencies.*.package` to disable or customize this behavior, respectively.
+  '';
+
   # TODO: added 2024-09-03 remove after 24.11
-  deprecateExtraOptions = true;
-  optionsRenamedToSettings = [
-    "configPath"
-    "binPath"
-    [
-      "api"
-      "host"
-    ]
-    [
-      "api"
-      "port"
-    ]
-    [
-      "tools"
-      "uname"
-    ]
-    [
-      "tools"
-      "uuidgen"
-    ]
-    [
-      "tools"
-      "curl"
-    ]
-    [
-      "tools"
-      "gzip"
-    ]
-    [
-      "tools"
-      "languageServer"
-    ]
-    "wrapper"
-  ];
+  inherit (import ./deprecations.nix) deprecateExtraOptions optionsRenamedToSettings;
 
   # Register nvim-cmp association
   imports = [
     { cmpSourcePlugins.codeium = "codeium-nvim"; }
   ];
 
-  settingsOptions = {
-    config_path = defaultNullOpts.mkStr {
-      __raw = "vim.fn.stdpath('cache') .. '/codeium/config.json'";
-    } "The path to the config file, used to store the API key.";
-
-    bin_path = defaultNullOpts.mkStr {
-      __raw = "vim.fn.stdpath('cache') .. '/codeium/bin'";
-    } "The path to the directory where the Codeium server will be downloaded to.";
-
-    api = {
-      host = defaultNullOpts.mkStr "server.codeium.com" ''
-        The hostname of the API server to use.
-      '';
-
-      port = defaultNullOpts.mkNullableWithRaw' {
-        # TODO: Added 2024-09-05; remove after 24.11
-        type = with lib.types; lib.nixvim.transitionType ints.positive toString (strMatching "[0-9]+");
-        pluginDefault = "443";
-        description = ''
-          The port of the API server to use.
-        '';
-      };
-    };
-
-    tools = {
```

Comment:
I wonder if a `settings.tools` option(s) should remain, as it is a sensible place to document and/or implement the dependency-based stuff?

Reply:
I feel like `description` and `settingsExample` are good places to document usages without having to maintain options themselves.

Reply:
> Also, the packages will be in `$PATH` so setting these options is pretty much redundant anyway.

Hm. If the plugin's default behaviour is to get the exe from PATH does defining the settings options achieve anything? If there's no benefit to it, I'd rather not do it :grin: